doc/source/tutorial/stats.rst

```diff
@@ -1190,3 +1191,246 @@ because the p-value is very low and the MGC test statistic is relatively high.
+points.
+
+    >>> engine.reset()
+    >>> engine.random(5)
+    array([[0.22166437, 0.07980522],
+           [0.72166437, 0.93165708],
+           [0.47166437, 0.41313856],
+           [0.97166437, 0.19091633],
+           [0.01853937, 0.74647189]])
+
+And here we advance the sequence to get the same second set of 5 points.
+
+    >>> engine.reset()
+    >>> engine.fast_forward(5)
+    >>> engine.random(5)
+    array([[0.51853937, 0.52424967],
+           [0.26853937, 0.30202745],
+           [0.76853937, 0.857583  ],
+           [0.14353937, 0.63536078],
+           [0.64353937, 0.01807683]])
+
+.. note:: By default, both :class:`~stats.qmc.Sobol` and
+   :class:`~stats.qmc.Halton` are scrambled. The convergence properties are
+   better, and it prevents the appearance of fringes or noticeable patterns
+   of points in high dimensions. There should be no practical reason not to
+   use the scrambled version.
+
+Making a QMC engine, i.e., subclassing ``QMCEngine``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To make your own :class:`~stats.qmc.QMCEngine`, a few methods have to be
+defined. Following is an example wrapping `numpy.random.Generator`.
+
+    >>> from scipy.stats import qmc
+    >>> class RandomEngine(qmc.QMCEngine):
+    ...     def __init__(self, d, seed=12345):
+    ...         self.d = d
+    ...         self.rng_seed = seed
+    ...         self.rng = np.random.default_rng(seed)
+    ...         self.num_generated = 0
+    ...
+    ...
+    ...     def random(self, n=1):
+    ...         self.num_generated += n
+    ...         return self.rng.random((n, self.d))
+    ...
+    ...
+    ...     def reset(self):
+    ...         self.rng = np.random.default_rng(self.rng_seed)
+    ...         self.num_generated = 0
+    ...         return self
+    ...
+    ...
+    ...     def fast_forward(self, n):
+    ...         self.random(n)
+    ...         return self
+
+Then we use it as any other QMC engine.
+
+    >>> engine = RandomEngine(2, seed=12345)
+    >>> engine.random(5)
+    array([[0.22733602, 0.31675834],
+           [0.79736546, 0.67625467],
+           [0.39110955, 0.33281393],
+           [0.59830875, 0.18673419],
+           [0.67275604, 0.94180287]])
+    >>> engine.reset()
+    >>> engine.random(5)
+    array([[0.22733602, 0.31675834],
+           [0.79736546, 0.67625467],
+           [0.39110955, 0.33281393],
+           [0.59830875, 0.18673419],
+           [0.67275604, 0.94180287]])
+
+Key takeaways
+^^^^^^^^^^^^^
+
+* QMC has rules! Be sure to read the documentation or you might have no
+  benefit over MC.
+* Use :class:`~stats.qmc.Sobol` if you need **exactly** :math:`2^m` points.
+  You can ask for more points, but you must follow some rules.
+* :class:`~stats.qmc.Halton` allows to sample, and skip an arbitrary number of
+  points.
+* Never remove the first points of the sequence. It will destroy the
+  properties.
+* Scrambling is always better.
```

Comment:
I would say that if I rename the section to something like Guidelines, this could stay. Or I can add something more about the need of scrambling with a bad example of unscrambled sequence. This would add a paragraph and a figure.

Reply:
I think it's fine as is, no need to make it longer.